plugin/url.vim

```diff
@@ -0,0 +1,25 @@
+command! UrlOpen call g:url#open(expand('<cWORD>'))
+
+function! g:url#open(word)
+  for transformer in g:url_transformers
+    let l:Try = function(transformer)
+    let result = l:Try(a:word)
+    if result != ""
+      call url#open_with_netrw(result)
+      return
+    endif
+  endfor
+  call url#open_with_netrw(a:word)
+endfunction
+
+function! g:url#open_with_netrw(url)
+  if has("patch-7.4.567")
+    call netrw#BrowseX(a:url, 0)
+  else
+    call netrw#NetrwBrowseX(a:url, 0)
+  endif
+endfunction
+
+if ! get(g:, 'url_opener_no_mappings', 0)
```

Comment:
If you wanted to get rid of `g:url_opener_no_mappings` and do the right thing automatically in 90% of cases (probably):

```vimscript
if maparg('gx', 'n') ==# '<Plug>NetrwBrowseX' || maparg('gx', 'n') ==# '<Plug>BrowseX'
  nnoremap gx :UrlOpen<CR>
endif
```

That will overwrite the default mapping, but it won't overwrite a custom mapping.